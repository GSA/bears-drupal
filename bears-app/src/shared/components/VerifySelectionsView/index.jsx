import PropTypes from 'prop-types'
import { Heading, Button } from '../index'

/**
 * afunctional component that renders a view of the form input state values
 * @component
 * @param {function} handleStepForward - an array of sections
 * @param {function} handleStepBack - determinate to render headings or not
 * @param {object} ui - translations
 * @param {array} data - inherited state of data in current session
 * @return {html} returns semantic html view for current input values
 */
const VerifySelectionsView = ({
  handleStepForward,
  handleStepBack,
  ui,
  data,
}) => {
  const { heading, buttonGroup } = ui
  return (
    <div>
      <Heading headingLevel={2}>{heading}</Heading>
      {data &&
        data.map((item, i) => {
          // get the sectional data
          // map all the criteria input legends and values
          const { section } = item
          return (
            <div key={i}>
              <Heading headingLevel={4} key={`section-${i}`}>
                {section.heading}
              </Heading>
              <div>
                {section.fieldsets.map((item, i) => {
                  const criteriaId = `criteria-${item.fieldset.criteriaKey}-${i}`
                  const result =
                    item.fieldset.inputs[0].inputCriteria.values.find(
                      x => x.selected
                    )
                  // if no value then return generic message
                  return result !== undefined ? (
                    <div>
                      <Heading headingLevel={5} key={`${criteriaId}`}>
                        {item.fieldset.legend}
                      </Heading>
                      {result.value}
                    </div>
                  ) : (
                    <div>
                      <Heading headingLevel={5} key={`${criteriaId}`}>
                        {item.fieldset.legend}
                      </Heading>
                      No input given.
                    </div>
                  )
                })}
              </div>
            </div>
          )
        })}
      <Button onClick={handleStepBack}>{buttonGroup[0].value}</Button>
      <Button onClick={handleStepForward}>{buttonGroup[1].value}</Button>
    </div>
  )
}

VerifySelectionsView.propTypes = {
  handleStepForward: PropTypes.func,
  handleStepBck: PropTypes.func,
  ui: PropTypes.object,
  data: PropTypes.array,
}

export default VerifySelectionsView
