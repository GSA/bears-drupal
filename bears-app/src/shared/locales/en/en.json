{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start",
      "list": [
        {
          "notice": "<div>Your information<strong>will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "heading": "Your potential benefits",
    "description": "<div>You must verify your eligibility and visit each agency to apply. <strong>Check with your city, state, or employer for other benefits.</strong></div>",
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    }
  },
  "notEligibleResults": {
    "heading": "Benefits you did not qualify for",
    "description": "<div>According to your answers you <strong>are not</strong> eligible for these benefits.</div>",
    "heading2": "Non-eligible Results",
    "description2": "Check with your state, city, or employer for other benefits."
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  }
}
