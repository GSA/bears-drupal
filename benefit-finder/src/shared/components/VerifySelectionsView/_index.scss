#benefit-finder {
  .verify-criteria-value {
    font-size: 20px;
    font-style: normal;
    font-weight: 700;
    line-height: 148.5%;
    margin-bottom: 30px;
    border-bottom: 1px solid lightgrey;
    padding-bottom: 30px;
  }

  .verify-criteria-legend {
    font-size: 16px;
    font-style: normal;
    font-weight: 400;
    line-height: 148.5%;
    padding-bottom: 8px;
  }

  .verify-criteria-section-heading {
    border-bottom: 2px solid #1a4480;
    padding-bottom: 30px;
  }
}
