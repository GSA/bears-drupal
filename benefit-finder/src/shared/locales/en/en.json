{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start:",
      "list": [
        {
          "notice": "<div>Your information<strong> will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "alertBanner": {
    "heading": "Error status",
    "description": "All fields marked required must be completed."
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "qualified": {
      "chevron": {
        "heading": "Your Custom List",
        "description": "<div><p><strong>Here is your list of potential Federal Benefits.</strong> Your state, city or employer may offer other benefits. Visit each agency providing the benefit to learn if you are truly eligible and conduct an actual application.</p></div>"
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you meet the criteria to qualify for these benefits:</strong></div>"
    },
    "notQualified": {
      "chevron": {
        "heading": "Benefits You Did Not Qualify For",
        "description": ""
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you did not meet the criteria to qualify for these benefits:</strong></div>"
    },
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    },
    "relativeBenefits": {
      "heading": "Other benefits that might be relevant to you"
    },
    "shareResults": {
      "heading": "Sharing and printing",
      "shareButton": "Share link",
      "emailButton": "Email results",
      "emailSubject": "Results from USAGovâ€™s benefit finder",
      "printButton": "Print results",
      "description": "Copy a link to this page with the criteria you selected above. Make sure to share it only with those you trust as your answers will be visible."
    }
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  },
  "sectionHeadings": {
    "start": "Complete Questions",
    "continue": "Questions Continued",
    "final": "Final Questions"
  },
  "select": {
    "labelSelect": "Select",
    "defaultValue": "Click to Select",
    "dateDefaultValue": "-Select-"
  },
  "date": {
    "alert": "Date should follow the format 01 - January 22 1977",
    "labelDay": "Day",
    "labelMonth": "Month",
    "labelYear": "Year",
    "monthOptions": [
      {
        "label": "01 - January"
      },
      {
        "label": "02 - February"
      },
      {
        "label": "03 - March"
      },
      {
        "label": "04 - April"
      },
      {
        "label": "05 - May"
      },
      {
        "label": "06 - June"
      },
      {
        "label": "07 - July"
      },
      {
        "label": "08 - August"
      },
      {
        "label": "09 - September"
      },
      {
        "label": "10 - October"
      },
      {
        "label": "11 - November"
      },
      {
        "label": "12 - December"
      }
    ]
  }
}
