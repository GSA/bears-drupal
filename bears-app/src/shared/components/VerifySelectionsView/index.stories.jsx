import VerifySelectionsView from './index.jsx'

export default {
  component: VerifySelectionsView,
  tags: ['autodocs'],
}

export const Primary = {}
