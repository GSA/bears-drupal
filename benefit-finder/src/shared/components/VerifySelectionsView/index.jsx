import { useEffect } from 'react'
import PropTypes from 'prop-types'
import parseDate from '../../utils/parseDate'
import * as apiCalls from '../../api/apiCalls'
import { Heading, Button } from '../index'

import './_index.scss'

/**
 * afunctional component that renders a view of the form input state values
 * @component
 * @param {function} handleStepForward - an array of sections
 * @param {function} handleStepBack - determinate to render headings or not
 * @param {object} ui - translations
 * @param {array} data - inherited state of data in current session
 * @return {html} returns semantic html view for current input values
 */
const VerifySelectionsView = ({
  handleStepForward,
  handleStepBack,
  ui,
  data,
}) => {
  const { stepIndicator, verifySelectionsView, buttonGroup } = ui
  const local = apiCalls.GET.Language()
  const dateFormatOptions = {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  }

  /**
   * afunctional component that renders markup when no value has been given
   * @component
   * @param {object} item - fieldset object with no selected value
   * @param {number} index - current index of mapped array
   * @return {html} returns html div
   */
  const NoInputGiven = ({ item, index }) => (
    <div className="verify-criteria-value">
      <div
        className="verify-criteria-legend"
        key={`criteria-${item.fieldset.criteriaKey}-${index}`}
      >
        {item.fieldset.legend}
      </div>
      No input given.
    </div>
  )

  /**
   * a functional component that renders markup for fieldset items with selected values
   * @component
   * @param {string} criteriaId - inherited string value
   * @param {string} legend - inherited string value
   * @param {object} selected - selected fieldset object value
   * @return {html} returns html div
   */
  const resultItem = ({ criteriaId, legend, selected }) => {
    return (
      <div className="verify-criteria-value">
        <div className="verify-criteria-legend" key={criteriaId}>
          {legend}
        </div>
        {typeof selected?.value === 'object'
          ? `${parseDate(selected.value).toLocaleDateString(
              local,
              dateFormatOptions
            )}`
          : selected?.value}
      </div>
    )
  }

  /**
   * a functional component that renders markup for fieldset items in the form
   * @component
   * @param {object} item - fieldset object
   * @param {number} index - current index of mapped array
   * @return {html} returns html div
   */
  const Items = ({ item, index }) => {
    return (
      <div>
        {resultItem({
          criteriaId: `criteria-${item.fieldset.criteriaKey}-${index}`,
          legend: item.fieldset.legend,
          selected: apiCalls.GET.SelectedValue(item),
        })}
        {apiCalls.GET.Children(item).map(childItem =>
          apiCalls.GET.SelectedValue(childItem) ? (
            resultItem({
              criteriaId: childItem.fieldset.criteriaKey,
              legend: childItem.fieldset.legend,
              selected: apiCalls.GET.SelectedValue(childItem),
            })
          ) : (
            <NoInputGiven
              item={childItem}
              key={childItem.fieldset.criteriaKey}
            />
          )
        )}
      </div>
    )
  }

  useEffect(() => {
    window.scrollTo(0, 0)
  }, [])

  return (
    <div className="verify-selections-view">
      <div className="grid-container">
        <Heading className="section-heading" headingLevel={1}>
          {verifySelectionsView?.heading}
        </Heading>
        <div className="section-wrapper">
          <div className="section">
            <Button
              className="step-back-link"
              onClick={handleStepBack}
              unstyled
            >
              {stepIndicator?.stepBackLink}
            </Button>
            <div>
              {data &&
                data.map((item, i) => {
                  // get the sectional data
                  // map all the criteria input legends and values
                  const { section } = item
                  return (
                    <div className="verify-criteria-section" key={i}>
                      <Heading
                        className="verify-criteria-section-heading"
                        headingLevel={2}
                        key={`section-${i}`}
                      >
                        {section.heading}
                      </Heading>
                      <div>
                        {section.fieldsets.map((item, i) => {
                          // if no value then return generic message
                          return apiCalls.GET.SelectedValue(item) ? (
                            <Items item={item} index={i} />
                          ) : (
                            <NoInputGiven item={item} index={i} />
                          )
                        })}
                      </div>
                    </div>
                  )
                })}
            </div>
            <Button secondary onClick={handleStepBack}>
              {buttonGroup[0].value}
            </Button>
            <Button onClick={handleStepForward}>{buttonGroup[1].value}</Button>
          </div>
        </div>
      </div>
    </div>
  )
}

VerifySelectionsView.propTypes = {
  handleStepForward: PropTypes.func,
  handleStepBck: PropTypes.func,
  ui: PropTypes.object,
  data: PropTypes.array,
}

export default VerifySelectionsView
