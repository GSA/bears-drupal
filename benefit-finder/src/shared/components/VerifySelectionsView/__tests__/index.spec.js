import renderer from 'react-test-renderer'
import VerifySelectionsView from '../index.jsx'
import * as en from '../../../locales/en/en.json'

describe('VerifySelectionsView', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(
      <VerifySelectionsView ui={en.verifySelectionsView} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
