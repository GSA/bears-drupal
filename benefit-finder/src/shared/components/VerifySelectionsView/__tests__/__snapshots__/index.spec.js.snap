// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifySelectionsView renders a match to the previous snapshot 1`] = `
<div
  className="verify-selections-view"
>
  <div
    className="grid-container"
  >
    <h1
      className=""
      id="skip-to-h1"
    />
    <div
      className="section-wrapper"
    >
      <div
        className="section"
      >
        <button
          className=""
          type="button"
        />
        <div />
        <div
          className="section-nav-btn-group"
        >
          <button
            className=""
            type="button"
          >
            Confirm
          </button>
          <button
            className=""
            type="button"
          >
            Back
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
